core/domain/collection_domain.py

```diff
@@ -1111,28 +1117,54 @@ def delete_node(self, exploration_id):
         del self.nodes[node_index]
 
     def add_skill(self, skill_name):
-        """Adds the new skill domain object with the specified name."""
+        """Adds the new skill domain object with the specified name.
+
+        Args:
+            skill_name: str. The name of the skill.
 
-        for _, skill in self.skills.iteritems():
-            if skill.name == skill_name:
-                raise ValueError(
-                    'Skill with name "%s" already exists.' % skill_name)
+        Returns
+            str. The id of the new skill.
+        """
+        if any(
```

Comment:
For formatting, prefer:

        if any([
            skill_name == skill.name for skill in self.skills.itervalues()]):

or

        if any([
            skill_name == skill.name
            for skill in self.skills.itervalues()]):

Ditto below.

Reply:
Updated.